### DISCUSSION TITLE
Add Custom attributes to Contacts

### DISCUSSION THREAD

**Discussion by Rishabhh17** [chatwoot/chatwoot#5674] [Self hosted]
Hello @pranavrajs @sojan-official @vishnu-narayanan .
1- I am using custom attributes like Country and City in the prechat. It is getting stored in the conversation pannel but not getting stored in the Contacts automatically.

2- I have also added the checkbox using custom attribute and marked it required but in widget ,without ticking also the conversation is getting started. I don't know where i am going wrong. Please help......

![image](https://user-images.githubusercontent.com/87717590/196664608-7488f4bc-08f9-4493-b858-6b51ec28013f.png)

![image](https://user-images.githubusercontent.com/87717590/196665888-e92748f4-2ee1-4263-8938-58066e640af4.png)

**Comment by muhsin-k** [comment]
> Hello @pranavrajs @sojan-official @vishnu-narayanan . 1- I am using custom attributes like Country and City in the prechat. It is getting stored in the conversation pannel but not getting stored in the Contacts automatically.

At the moment, we don't support this. We will discuss in the team and will prioritize in the upcoming releases.

> 2- I have also added the checkbox using custom attribute and marked it required but in widget ,without ticking also the conversation is getting started. I don't know where i am going wrong. Please help......

We have fixed this issue in https://github.com/chatwoot/chatwoot/pull/5648 and will be available in the next release (v2.11).
> 
> ![image](https://user-images.githubusercontent.com/87717590/196664608-7488f4bc-08f9-4493-b858-6b51ec28013f.png)
> 
> ![image](https://user-images.githubusercontent.com/87717590/196665888-e92748f4-2ee1-4263-8938-58066e640af4.png)

**Reply by Rishabhh17** [reply]
@muhsin-k When are you releasing the updated version?

**Comment by muhsin-k** [comment]
Next month. You could use the develop branch since the fix is already merged. [Reference](https://github.com/chatwoot/chatwoot/commit/444809cc683b4b107569d5194440cc59c38b17a8)

**Reply by Rishabhh17** [reply]
Hi @muhsin-k Can you please guide me how to use the develop branch. I am using the selfhosted installation with AWS Marketplace chatwoot instance. It would be great if you can guide me.
